src/emulator/gui/include/gui/state.h

```diff
@@ -93,7 +93,9 @@ struct State {
     char disassembly_count[5] = "100";
     std::vector<std::string> disassembly;
 
-    GLObject background_texture;
+    GLuint current_background = 0;
+    std::map<std::string, GLObject> game_backgrounds;
+    std::map<std::string, GLObject> user_backgrounds;
```

Comment:
You probably don't need a std::map for `user_backgrounds`. 😓 When you set a new background, there isn't any reason to keep the old background in memory (you probably aren't going to use it again).

But I see why you did it, you can't change GLObject once it was created (init freaks out). I would add a method to GLObject so we can replace the texture (deletes old, sets new), but if someone else has a better idea, please tell.

Reply:
For me here is juste better  and is need obligatoty separate game and user background for all works